index.js

```diff
@@ -53,19 +40,29 @@ export function createHandler (functions) {
 
     self.addEventListener('message', function (event) {
       const {command, id, message} = event.data
-      Promise
-        .resolve(functions[command].call(null, cache, message))
-        .then(results => {
-          self.postMessage({
-            command,
-            id,
-            message: results
-          })
-        })
-        .catch(e => {
-          e.id = id
-          throw e
+
+      let results
+      try {
+        results = functions[command].call(null, cache, message)
```

Comment:
@mattwigway so I used `Promise.resolve` to allow any function to return a Promise and therefore be `async` in nature. But `Promise.resolve` has a [different behavior](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/resolve) when dealing with non-Promises than I intented. What I really wanted was the behavior of `await`. Since you can `await` a non-promise and if it throws an error, it will propagate. So let's `await` this function call.